src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetailsTests.cs

```diff
@@ -495,6 +495,23 @@ End Class
 ");
         }
 
+        [Fact]
+        public async Task CA2200_NoDiagnosticsForFieldInClassOrRecord()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+public class C
+{
+    private string s = default(string) ?? throw new Exception();
+}
+
+public record R
+{
+    private string s = default(string) ?? throw new Exception();
+}");
+        }
```

Comment:
```suggestion
        [Theory]
        [InlineData("class")]
        [InlineData("struct")]
        [InlineData("record")]
        public async Task CA2200_NoDiagnosticsForFieldThrowExpression(string typeKind)
        {
            await VerifyCS.VerifyAnalyzerAsync(@"
using System;

public " + typeKind + @" C
{
    private string s = default(string) ?? throw new Exception();
}");
        }
```

Reply:
@Evangelink I think it won't work with structs that way. It will result in a compile error.